src/rules.c

```diff
@@ -4135,6 +4141,29 @@ solver_addrecommendsrules(Solver *solv)
   solv->recommendsrules_end = solv->nrules;
 }
 
+/* add rules to exclude solvables provided by lower
+ * precedence repositories */
+void solver_addstrictrepopriorules(struct s_Solver *solv)
+{
+  Pool *pool = solv->pool;
+  Solvable *s;
+  Id sid, p, pp;
+
+  solv->strictrepopriorules = solv->nrules;
+  FOR_POOL_SOLVABLES(sid)
+  {
+	s = pool->solvables + sid;
+	int max_prio = 0;
```

Comment:
if (s->repo == pool->installed) continue;

Reply:
thx for catching this